internal/monitors/conviva/conviva.go

```diff
@@ -0,0 +1,276 @@
+// [here](https://github.com/signalfx/integrations/tree/master/conviva/docs) which correspond to the Conviva
+// `metric parameters` [here](https://community.conviva.com/site/global/apis_data/experience_insights_api/index.gsp#metrics)
+// Where an account is not provided the default account is used. Where no filters are specified the
+// `All Traffic` filter is used. Where MetricLens dimensions are not specified all MetricLens dimensions
+// are used. The `_ALL_` keyword means all. Dimensions only apply to MetricLenses. If specified for a
+// regular metric they will be ignored. MetricLens dimensions listed in `excludeMetricLensDimensions` will be excluded.
+//
+// ```
+//monitors:
+//- type: conviva
+//  pulseUsername: <username>
+//  pulsePassword: <password>
+//  metricConfigs:
+//    - account: c3.NBC
+//      metric: quality_metriclens
+//      filters:
+//        - All Traffic
+//      metricLensDimensions:
+//        - Cities
+//    - metric: avg_bitrate
+//      filters:
+//        - _ALL_
+//    - metric: concurrent_plays
+//    - metric: audience_metriclens
+//      filters:
+//        - All Traffic
+//      metricLensDimensions:
+//        - _ALL_
+//      excludeMetricLensDimensions:
+//        - CDNs
+// ```
+//
+// Add the extra dimension metric_source as shown in sample configuration below for the convenience of searching
+// for your metrics in SignalFx using the metric_source value you specify.
+//
+// ```
+//monitors:
+//- type: conviva
+//  pulseUsername: <username>
+//  pulsePassword: <password>
+//  extraDimensions:
+//    metric_source: conviva
+// ```
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+const (
+	metricURLFormat     = "https://api.conviva.com/insights/2.4/metrics.json?metrics=%s&account=%s&filter_ids=%s"
+	metricLensURLFormat = metricURLFormat + "&metriclens_dimension_id=%d"
+)
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig
+	// Conviva Pulse username required with each API request.
+	Username string `yaml:"pulseUsername" validate:"required"`
+	// Conviva Pulse password required with each API request.
+	Password       string `yaml:"pulsePassword" validate:"required" neverLog:"true"`
+	TimeoutSeconds int    `yaml:"timeoutSeconds" default:"10"`
+	// Conviva metrics to fetch. The default is quality_metriclens metric with the "All Traffic" filter applied and all quality_metriclens dimensions.
+	MetricConfigs []*metricConfig `yaml:"metricConfigs"`
+}
+
+// Monitor for conviva metrics
+type Monitor struct {
+	Output  types.Output
+	cancel  context.CancelFunc
+	ctx     context.Context
+	client  httpClient
+	timeout time.Duration
+}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Configure monitor
+func (m *Monitor) Configure(conf *Config) error {
+	if conf.MetricConfigs == nil {
+		conf.MetricConfigs = []*metricConfig{{Metric: "quality_metriclens"}}
+	}
+	for _, metricConf := range conf.MetricConfigs {
```

Comment:
Get rid of this after you make mutex a value on the struct instead of a pointer.